src/lgamma.rs

```diff
@@ -0,0 +1,97 @@
+use std::f64::{consts::LN_2, INFINITY, NAN};
+
+use log::warn;
+
+use crate::{gammafn, sinpi};
+
+extern "C" {
+    pub fn lgammacor(x: f64) -> f64;
+}
+
+/// Machine dependent constants for IEEE double precision
+const XMAX: f64 = 2.5327372760800758e+305;
+const DXREL: f64 = 1.490116119384765625e-8;
+
+/// The function lgammafn computes log|gamma(x)|.  The function
+/// lgammafn_sign in addition assigns the sign of the gamma function
+/// to the address in the second argument if this is not NULL.
+///
+/// ## NOTES
+///
+/// This routine is a translation into C of a Fortran subroutine
+/// by W. Fullerton of Los Alamos Scientific Laboratory.
+///
+/// The accuracy of this routine compares (very) favourably
+/// with those of the Sun Microsystems portable mathematical
+/// library.
+///
+/// ./toms708.c  has  gamln()
+pub fn lgammafn(x: f64) -> f64 {
+    lgammafn_sign(x, None)
+}
+
+/// Compute the log gamma function and its sign.
+///
+/// This function computes the logarithm of the absolute value of the gamma function of x
+/// and also sets the sign of the gamma function in `sgn`.
+fn lgammafn_sign(x: f64, sgn: Option<&mut i32>) -> f64 {
+    if let Some(sgn) = sgn {
+        *sgn = 1;
+        if x < 0.0 && ((-x).floor() % 2.0) == 0.0 {
+            *sgn = -1;
+        }
+    }
+
+    if x.is_nan() {
+        return NAN;
+    }
+
+    if x <= 0.0 && x == x.trunc() {
+        // Negative integer argument
+        return INFINITY; // +Inf, since lgamma(x) = log|gamma(x)|
+    }
+
+    let y = x.abs();
+
+    if y < 1e-306 {
+        return -y.ln();
+    }
+    if y <= 10.0 {
+        return gammafn(x).abs().ln();
+    }
+
+    // y = |x| > 10
+    if y > XMAX {
+        return INFINITY;
+    }
+
+    if x > 0.0 {
+        // Positive x
+        if x > 1e17 {
+            return x * (x.ln() - 1.0);
+        } else if x > 4934720.0 {
+            return LN_2 + (x - 0.5) * x.ln() - x;
```

Comment:
```suggestion
            return M_1_SQRT_2PI + (x - 0.5) * x.ln() - x;
```
Shouldn't it be `M_1_SQRT_2PI`? And add the following to `src/rmath` (from `Rmath.h`):
```rust
pub const M_1_SQRT_2PI 0.918938533204672741780329736406 // log(sqrt(2*pi))
```

Same for line 75.

Reply:
Actually is `M_LN_SQRT_2PI` and yeah fixed. Thanks!